src/hotspot/cpu/riscv/templateInterpreterGenerator_riscv.cpp

```diff
@@ -296,7 +296,8 @@ address TemplateInterpreterGenerator::generate_math_entry(AbstractInterpreter::M
       ;
   }
   if (entry_point != nullptr) {
-    __ jr(continuation);
+    __ mv(t1, continuation);
```

Comment:
I see `continuation` will be changed into `x9` for some switch cases, so maybe we can let `continuation` be `x9` and unconditionallly move `ra` into `x9` on entry. That would save the update of `continuation` and avoid use of `t1`, which I think will be cleaner.

Reply:
Hmm ... I don't understand why we would prefer `t1` here. It's actually a RET here, right? So I think that we should prefer `ra` for `jr` instead which indicates a RAS Pop. I mean something like this: 
```
if (continuation != ra) {
  __ mv(ra, continuation);
}
__ jr(ra);
```

Reply:
I was fooled by the use of Jump Register mnemonic, this is a **jump** and cannot be used with **ra**.
So yes this is a return and should use the **ret** mnemonic.
Fixed.